### PR TITLE
Adding useNewsletter hook

### PR DESCRIPTION
**Pull request by lucis**
[PR Marin](https://github.com/vtex-sites/marinbrasil.store/pull/148)
[DP Marin](https://preview-148--marinbrasil.vtex.app/)

### CODE REVIEW

File: packages/gatsby-theme-vtex/src/sdk/newsletter/useNewsletter.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    },
+    method: 'PATCH',
+    body: JSON.stringify(data),
+  }).then((response) => {
+    if (response.ok) {
+      return response.json()
+    }
+
+    console.error(response)
+    throw new Error()
```
Comment (lariciamota) [inline packages/gatsby-theme-vtex/src/sdk/newsletter/useNewsletter.ts:23]: What about adding some description to this error?

Comment (lucis) [reply packages/gatsby-theme-vtex/src/sdk/newsletter/useNewsletter.ts:23]: Yeah, unfortunately we have no way to guarantee how our API errors come... That's why I've only thrown one Error and let the caller decide the error message.

Although it's highly unlikely that any error message from the Master Data would be directed to the user

File: packages/gatsby-theme-vtex/src/sdk/newsletter/useNewsletter.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      accept: 'application/vnd.vtex.ds.v10+json',
+    },
+    method: 'PATCH',
+    body: JSON.stringify(data),
+  }).then((response) => {
+    if (response.ok) {
+      return response.json()
+    }
+
+    console.error(response)
```
Comment (tlgimenes) [inline packages/gatsby-theme-vtex/src/sdk/newsletter/useNewsletter.ts:22]: please remove console before merging

File: packages/gatsby-theme-vtex/src/sdk/newsletter/useNewsletter.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  const [error, setError] = useState(false)
+
+  const addUser = useCallback(
+    (user: { name: string; email: string }) => {
+      setError(false)
+      setData(null)
+      setLoading(true)
+
+      return addToNewsletter(user)
+        .then((response) => {
```
Comment (tlgimenes) [inline packages/gatsby-theme-vtex/src/sdk/newsletter/useNewsletter.ts:38]: why not using `await` constructor ?

**Review by tlgimenes — APPROVED** [review]

**Review by lariciamota — APPROVED** [review]

**Pull request merged**